src/PotentialBond.h

```diff
@@ -0,0 +1,391 @@
+
+    PDataFlags flags = this->m_pdata->getFlags();
+    bool compute_virial = flags[pdata_flag::pressure_tensor];
+
+    Scalar bond_virial[6];
+    for (unsigned int i = 0; i < 6; i++)
+        bond_virial[i] = Scalar(0.0);
+
+    ArrayHandle<typename Bonds::members_t> h_bonds(m_bond_data->getMembersArray(),
+                                                   access_location::host,
+                                                   access_mode::read);
+    ArrayHandle<typeval_t> h_typeval(m_bond_data->getTypeValArray(),
+                                     access_location::host,
+                                     access_mode::read);
+
+    unsigned int max_local = m_pdata->getN() + m_pdata->getNGhosts();
+
+    // for each of the bonds
+    const unsigned int size = (unsigned int)m_bond_data->getN();
+
+    for (unsigned int i = 0; i < size; i++)
+        {
+        // lookup the tag of each of the particles participating in the bond
+        const typename Bonds::members_t& bond = h_bonds.data[i];
+        assert(bond.tag[0] < m_pdata->getMaximumTag() + 1);
+        assert(bond.tag[1] < m_pdata->getMaximumTag() + 1);
+
+        // transform a and b into indices into the particle data arrays
+        // (MEM TRANSFER: 4 integers)
+        unsigned int idx_a = h_rtag.data[bond.tag[0]];
+        unsigned int idx_b = h_rtag.data[bond.tag[1]];
+
+        // throw an error if this bond is incomplete
+        if (idx_a >= max_local || idx_b >= max_local)
+            {
+            std::ostringstream stream;
+            stream << "Error: bond " << bond.tag[0] << " " << bond.tag[1] << " is incomplete.";
+            throw std::runtime_error(stream.str());
+            }
+
+        // calculate d\vec{r}
+        // (MEM TRANSFER: 6 Scalars / FLOPS: 3)
+        Scalar3 posa = make_scalar3(h_pos.data[idx_a].x, h_pos.data[idx_a].y, h_pos.data[idx_a].z);
+        Scalar3 posb = make_scalar3(h_pos.data[idx_b].x, h_pos.data[idx_b].y, h_pos.data[idx_b].z);
+
+        // access charge (if needed)
+        Scalar charge_a = Scalar(0.0);
+        Scalar charge_b = Scalar(0.0);
+        if (evaluator::needsCharge())
+            {
+            charge_a = h_charge.data[idx_a];
+            charge_b = h_charge.data[idx_b];
+            }
+
+        // Commenting out to avoid using minimum image for bonds
+        // // if the vector crosses the box, pull it back
+        // dx = box.minImage(dx);
+
+        // Get the box lengths and tilt factors
+        Scalar3 lengths = box.getL();
+
+        // get images of particles
+        int3 images_a = make_int3(0, 0, 0);
+        int3 images_b = make_int3(0, 0, 0);
+        if (idx_a >= m_pdata->getN())
+            images_a = m_pdata->getImages(bond.tag[0]);
+        if (idx_b >= m_pdata->getN())
+            images_b = m_pdata->getImages(bond.tag[1]);
+
+        // unwrap particles
+        posa += images_a * lengths;
+        posb += images_b * lengths;
+
+        Scalar3 dx = posa - posb;
```

Comment:
I think this should be the right place to unwrap the positions and get the bond vector.